integration-tests/src/tests/client/features/stateless_validation.rs

```diff
@@ -341,3 +342,67 @@ fn test_chunk_state_witness_bad_shard_id() {
     tracing::info!(target: "test", "error message: {}", error_message);
     assert!(error_message.contains("shard"));
 }
+
+/// Test that block approvals can serve as chunk endorsements by not sending chunk endorsements from block producers
```

Comment:
nit. two things not clear to me, 1) what is missing in the test that causes chunk endorsements not propagates (I guess it is
`let output = env.propagate_chunk_state_witnesses(false); env.propagate_chunk_endorsements(false);` in the first test above, maybe indicate this.
2) what indicates that this test is successful, is it the checking the chunks at line 396?

Reply:
Added comments to explain